### PR TITLE
build_url added to all API v3 build endpoints

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/readthedocs/readthedocs.org/issues/7361

### CODE REVIEW

File: readthedocs/api/v3/tests/responses/projects-builds-detail.json [hunk 1]
```diff
@@ -10,6 +10,11 @@
         "project": "https://readthedocs.org/api/v3/projects/project/",
         "version": "https://readthedocs.org/api/v3/projects/project/versions/v1.0/"
     },
+    "urls": {
+        "build": "https://readthedocs.org/projects/project/builds/1/",
+        "project": "https://readthedocs.org/projects/project/",
+        "version": "https://readthedocs.org/dashboard/project/version/v1.0/"
```
Comment (stsewd) [inline readthedocs/api/v3/tests/responses/projects-builds-detail.json:16]: The project and version URL are going to be the same on every element of the list, not sure about that (we do the same in _links already). I think maybe just add the build url for now, and see if we need the others later.

Comment (saadmk11) [reply readthedocs/api/v3/tests/responses/projects-builds-detail.json:16]: Okay sure! updated :)

**Comment by saadmk11** [comment]
I think this https://github.com/readthedocs/readthedocs.org/issues/7296 is related

File: readthedocs/api/v3/tests/responses/projects-builds-detail.json [hunk 2]
```diff
@@ -10,6 +10,9 @@
         "project": "https://readthedocs.org/api/v3/projects/project/",
         "version": "https://readthedocs.org/api/v3/projects/project/versions/v1.0/"
     },
+    "urls": {
+        "build": "https://readthedocs.org/projects/project/builds/1/"
+    },
```
Comment (humitos) [inline readthedocs/api/v3/tests/responses/projects-builds-detail.json:15]: What about showing the same URLs we are listing in `_links`?

* `_self`
* `project`
* `version`

I think it makes sense to keep that matching between the two. `_links` refers to API URLs and `urls` refer to user/dashboard URLs. Both with the same naming.

**Review by humitos — COMMENTED** [review]
Made a suggestion to match the same structure than `_links`.

File: readthedocs/api/v3/tests/responses/projects-builds-detail.json (hunk 2 above)
Comment (saadmk11) [reply readthedocs/api/v3/tests/responses/projects-builds-detail.json]: @humitos  Yeah! That makes sense :+1: . Updated the PR

File: readthedocs/api/v3/serializers.py [hunk 3]
```diff
@@ -88,6 +88,31 @@ def get_project(self, obj):
         return self._absolute_url(path)
 
 
+class BuildURLsSerializer(BaseLinksSerializer, serializers.Serializer):
+    _self = serializers.URLField(source='get_full_url')
```
Comment (stsewd) [inline readthedocs/api/v3/serializers.py:92]: we are not using self in the rest of url serializers, I think we can leave it as `build`

Comment (saadmk11) [reply readthedocs/api/v3/serializers.py]: sure!

**Review by humitos — APPROVED** [review]
Thanks for your contribution! :tada:

**Pull request merged**